browser/js/dashboard/dashboard.controller.js

```diff
@@ -23,8 +24,13 @@ app.controller('dashboardCtrl', ['$scope', 'DashboardFactory', '$rootScope', 'da
     // The resolve fetched a list of dashboards for this user.  Note that this
     // does not entail the graphs, which must be fetched later.
     $scope.dashboardList = dashList;
-    if (dashList.length > 0)
-        $scope.selectedDb = dashList[0];
+    if (dashList.length > 0){
+        let indexOfDashToLoad = 0;
+        if($stateParams.dashToLoad){
+            indexOfDashToLoad = DashboardFactory.findIndexToLoad(dashList, $stateParams.dashToLoad.name)
+        }
+        $scope.selectedDb = dashList[indexOfDashToLoad];
+    }
```

Comment:
Might be tidier to use a series of resolves to get the dashToLoad, too:

1. Resolve the dashList.
2. If $stateParams.dashToLoad && dashlist.length, resolve the indexOfDashToLoad. Else resolve null.
3. If indexOfDashToLoad !== null, resolve the dashToLoad by returning dashList[indexOfDashToLoad]